sources/osgShadow/ShadowMapAtlas.js

```diff
@@ -0,0 +1,600 @@
+var ShadowTextureAtlas = require( 'osgShadow/ShadowTextureAtlas' );
+var ShadowMap = require( 'osgShadow/ShadowMap' );
+
+/**
+ *  ShadowMapAtlas provides an implementation of shadow textures.
+ * here, one shadow
+ *  @class ShadowMapAtlas
+ */
+var ShadowMapAtlas = function ( settings ) {
+
+    this._lights = [];
+    this._shadowMaps = [];
+    this._viewportDimension = [];
+
+    ShadowTechnique.apply( this, arguments );
+
+    this._texture = new ShadowTextureAtlas();
+    this._textureUnitBase = 4;
+    this._textureUnit = this._textureUnitBase;
+
+    // see shadowSettings.js header for param explanations
+    this._textureMagFilter = undefined;
+    this._textureMinFilter = undefined;
+
+
+    this._textureSize = 1024;
+    this._shadowMapSize = 256;
+
+    this._receivingStateset = undefined;
+
+    this._shaderProcessor = undefined;
+
+    if ( settings ) {
+
+        this.setShadowSettings( settings );
+        if ( settings.atlasSize ) this._textureSize = settings.atlasSize;
+        if ( settings.textureSize ) this._shadowMapSize = settings.textureSize;
+
+    }
+
+    this._texelSizeUniform = Uniform.createFloat1( 1.0 / this._textureSize, 'texelSize' );
+
+    var unifRenderSize = Uniform.createFloat2( 'RenderSize' );
+    this._renderSize = unifRenderSize.getInternalArray();
+    this._renderSize[ 0 ] = this._renderSize[ 1 ] = this._textureSize;
+
+    this._numShadowWidth = this._textureSize / this._shadowMapSize;
+    this._numShadowHeight = this._textureSize / this._shadowMapSize;
+
+};
+
+
+/** @lends ShadowMapAtlas.prototype */
+ShadowMapAtlas.prototype = MACROUTILS.objectLibraryClass( MACROUTILS.objectInherit( ShadowTechnique.prototype, {
+
+    getDepthRange: function ( numShadow ) {
+        return this._shadowMaps[ numShadow ].getDepthRange();
+    },
+
+    getTexture: function () {
+        return this._texture;
+    },
+
+    isDirty: function ( numShadow ) {
+        if ( numShadow !== undefined ) {
+            return this._shadowMaps[ numShadow ].isDirty();
+        } else {
+            for ( var i = 0, l = this._shadowMaps.length; i < l; i++ ) {
+                if ( this._shadowMaps[ i ].isDirty() ) return true;
+            }
+        }
+        return false;
+    },
+    /**
+     * at which Texture unit number we start adding texture shadow
+     */
+    setTextureUnitBase: function ( unitBase ) {
+        this._textureUnitBase = unitBase;
+        this._textureUnit = unitBase;
+
+        for ( var i = 0, l = this._shadowMaps.length; i < l; i++ ) {
+            this._shadowMaps[ i ].setTextureUnitBase( unitBase );
+        }
+    },
+
+    /* Sets  shadowSettings
+     */
+    setShadowSettings: function ( shadowSettings ) {
+
+        if ( !shadowSettings )
+            return;
+
+        for ( var i = 0, l = this._shadowMaps.length; i < l; i++ ) {
+            this._shadowMaps[ i ].setShadowSettings( shadowSettings );
+        }
+        this.setTextureSize( shadowSettings.textureSize );
```

Comment:
We agree that we can't really resize anything after the init is set. So probably we should remove this call.